src/controllers/keyboard/keyboardeventfilter.cpp

```diff
@@ -119,53 +127,67 @@ bool KeyboardEventFilter::eventFilter(QObject*, QEvent* e) {
     } else if (e->type() == QEvent::KeyboardLayoutChange) {
         // This event is not fired on ubunty natty, why?
         // TODO(XXX): find a way to support KeyboardLayoutChange Bug #997811
-        //qDebug() << "QEvent::KeyboardLayoutChange";
-    }
-    return false;
-}
-
-QKeySequence KeyboardEventFilter::getKeySeq(QKeyEvent* e) {
-    QString modseq;
-    QKeySequence k;
 
-    // TODO(XXX) check if we may simply return QKeySequence(e->modifiers()+e->key())
+        // TODO(Tomasito) Event is not fired when Mixxx has no focus. So, if a user loses focus on Mixxx,
+        // ...            changes the keyboard layout and gets back to Mixxx, the keyboard mapping will not
+        // ...            be translated to his current locale: Add a check on focusInEvent (on MixxxMainWindow)
 
-    if (e->modifiers() & Qt::ShiftModifier)
-        modseq += "Shift+";
+        static QString prevLayout;
```

Comment:
Please use a member for this purpose instead of a local static variable. The local static variable is shared between all instances of this class!